From b6198512858a2f9498dd1b8ecdf8f62a9385ad98 Mon Sep 17 00:00:00 2001
From: gbhakteX <gurudattax.bhakte@intel.com>
Date: Mon, 10 Sep 2012 16:18:38 +0530
Subject: [PORT FROM MAIN] Audio-MW: Playback support of MP2 contents in .avi container.

BZ: 55441

Currently only "Wave Format = 0x55 (MPEG Layer 3)" is supported in AVI container.
With this fix, adding support for "Wave Format = 0x50 (MPEG)".
As per "RFC 2361 - WAVE and AVI Codec Registries":
      A.45 MPEG
      WAVE form Registration Number (hex): 0x0050
      Codec ID in the IANA Namespace:      audio/vnd.wave;codec=50
      WAVE form wFormatTag ID:             WAVE_FORMAT_MPEG

Change-Id: I2f0da3967a9e018617720f185da88c943ec6b1a4
Orig-Change-Id: Ia8697b4eee27183c4396997fe4c2db335eeda397
Signed-off-by: gbhakteX <gurudattax.bhakte@intel.com>
---
 media/libstagefright/AVIExtractor.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/AVIExtractor.cpp b/media/libstagefright/AVIExtractor.cpp
index c4ac5ab..e48e9f8 100644
--- a/media/libstagefright/AVIExtractor.cpp
+++ b/media/libstagefright/AVIExtractor.cpp
@@ -711,7 +711,7 @@ status_t AVIExtractor::parseStreamFormat(off64_t offset, size_t size) {
     } else {
         uint32_t format = U16LE_AT(data);
 
-        if (format == 0x55) {
+        if ((format == 0x55) || (format == 0x50)) {
             track->mMeta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_AUDIO_MPEG);
         } else {
             ALOGW("Unsupported audio format = 0x%04x", format);
-- 
1.7.4.1

